### PR TITLE
BDR PoT support for Azure and GCloud

### PR DESCRIPTION
**Pull request by vibhorkum**
With this PR we are adding support for Azure and GCloud

### CODE REVIEW

File: edbdeploy/data/terraform/azure/environments/vm/vm.tf [hunk 1]
```diff
@@ -333,6 +480,43 @@ resource "azurerm_linux_virtual_machine" "pem_server" {
...
+    publisher = var.azure_publisher
+    offer     = var.azure_offer
+    sku       = var.azure_sku
+
+    version = "latest"
+  }
+
+  os_disk {
+    name                 = format("bdr-witness-%s-%s-%s", var.cluster_name, "EDB-VM-OS-Disk", count.index)
+    storage_account_type = var.pem_server["volume"]["storage_account_type"]
```
Comment (jt-edb) [inline edbdeploy/data/terraform/azure/environments/vm/vm.tf:513]: Wrong variable name, should be `var.bdr_witness_server["volume"]["storage_account_type"]`

File: edbdeploy/data/terraform/azure/main.tf [hunk 2]
```diff
@@ -80,7 +82,7 @@ module "vm" {
   add_hosts_filename              = var.add_hosts_filename
   pooler_type                     = var.pooler_type
   pooler_local                    = var.pooler_local
-  network_count                   = var.pooler_server["count"] > var.postgres_server["count"] ? var.pooler_server["count"] : var.postgres_server["count"]
+  network_count                   = max(var.pooler_server["count"], var.postgres_server["count"], var.bdr_server["count"])
```
Comment (jt-edb) [inline edbdeploy/data/terraform/azure/main.tf:85]: This way is much more elegant! 👍

**Review by jt-edb — COMMENTED** [review]
LGTM. A few minor comments.

File: edbdeploy/data/terraform/azure/environments/vm/vm.tf [hunk 3]
```diff
@@ -82,6 +84,52 @@ resource "azurerm_network_interface" "postgres_public_nic" {
...
+}
+
+resource "azurerm_network_interface" "bdr_public_nic" {
+  count               = var.bdr_server["count"]
+  name                = format("bdr-%s-%s-%s", var.cluster_name, "edb_public_nic", count.index)
+  resource_group_name = var.resourcegroup_name
+  location            = var.azure_region
+
+  ip_configuration {
+    name      = "PG_Private_Nic_${count.index}"
```
[the remark this answers is not in the document]
Comment (vibhorkumar123) [reply edbdeploy/data/terraform/azure/environments/vm/vm.tf]: Fixed

File: edbdeploy/data/terraform/azure/environments/vm/vm.tf [hunk 4]
```diff
@@ -82,6 +84,52 @@ resource "azurerm_network_interface" "postgres_public_nic" {
...
+}
+
+resource "azurerm_network_interface" "bdr_witness_public_nic" {
+  count               = var.bdr_witness_server["count"]
+  name                = format("bdr-witness-%s-%s-%s", var.cluster_name, "edb_public_nic", count.index)
+  resource_group_name = var.resourcegroup_name
+  location            = var.azure_region
+
+  ip_configuration {
+    name      = "PG_Private_Nic_${count.index}"
```
[the remark this answers is not in the document]
Comment (vibhorkumar123) [reply edbdeploy/data/terraform/azure/environments/vm/vm.tf]: ixed

File: edbdeploy/data/terraform/azure/environments/vm/vm.tf [hunk 5]
```diff
@@ -333,6 +480,43 @@ resource "azurerm_linux_virtual_machine" "pem_server" {
   tags = var.project_tags
 }
 
+resource "azurerm_linux_virtual_machine" "bdr_witness_server" {
+  count               = var.bdr_witness_server["count"]
+  name                = format("%s-%s%s", var.cluster_name, "bdr-witness", count.index + 1)
+  resource_group_name = var.resourcegroup_name
+  location            = var.azure_region
+
+  size           = var.pem_server["instance_type"]
```
[the remark this answers is not in the document]
Comment (vibhorkumar123) [reply edbdeploy/data/terraform/azure/environments/vm/vm.tf:489]: Fixed

File: edbdeploy/data/terraform/azure/environments/vm/vm.tf (hunk 1 above)
Comment (vibhorkumar123) [reply edbdeploy/data/terraform/azure/environments/vm/vm.tf]: Fixed

File: edbdeploy/data/terraform/azure/main.tf (hunk 2 above)
Comment (vibhorkumar123) [reply edbdeploy/data/terraform/azure/main.tf:85]: Thank you!

File: edbdeploy/data/terraform/azure/environments/vm/vm.tf (hunk 5 above)
[the remark this answers is not in the document]
Comment (vibhorkumar123) [reply edbdeploy/data/terraform/azure/environments/vm/vm.tf]: Fixed

**Review by jt-edb — APPROVED** [review]
LGTM

**Pull request merged**